### ISSUE TITLE
We lie about fetching the next exercise...

### ISSUE THREAD

**Issue by kytrinyx** [exercism/cli#187]
In `cmd/submit.go` we say _To get the next exercise, run "exercism fetch" again_ whenever you submit. This is not actually true. We only unlock the next exercise if this is the first iteration of a submission.

I've modified the response from the server to include an `iteration` key, which is an `int`. When `iteration` is 1, then we can say that a new problem is available.

... And of course even that might be a lie, because if you're at the end of a track, we don't actually tell you that.

https://github.com/exercism/exercism.io/issues/2315

**Comment by Tonkpils** [comment]
I'm not 100% familiar with the design of the iterations on the server how feasible would it be to actually have a count of total exercises for a track and a count of submitted exercises for a user? That would allow us to tackle two issues. I can probably spend some time to take a look at it but I haven't been in the codebase in a while. Thought maybe you'd have a better idea.

**Comment by kytrinyx** [comment]
We can know how many issues there are in a track very easily--the Problem API already has that information and can send the number along when fetching.

The number of things they've completed is another matter--that's the exercism.io repo. But... what if every time they fetch we stick the name of the track/problem in a redis set? Or we could even do a redis hash of problem/track => fetch count, which might be interesting.

It's super cheap to do, it would keep it on the same server.

On the other hand, we don't have their user id. Hm. At that point we'd have to use their key to get their user ID, which sucks. So maybe it's easier to just make a call to the exercism.io api to get the number. We can make the two calls in parallel.

**Comment by kytrinyx** [comment]
BTW - I haven't deployed yet. I'm super jet-lagged and don't want to accidentally break something when I don't have the brain power to fix it. I'll deploy the `iterations` thing tomorrow.

**Comment by Tonkpils** [comment]
> On the other hand, we don't have their user id. Hm. At that point we'd have to use their key to get their user ID, which sucks. So maybe it's easier to just make a call to the exercism.io api to get the number. We can make the two calls in parallel.

What if the we stuck the submission count in a redis hash with the API key as the hash key that way we could just match the API key with the submission count on fetch. 

I don't want to clutter this issue with what seems to be a separate discussion. Want to move it to a more appropriate place?

**Comment by kytrinyx** [comment]
Yeah, let's move the discussion to a new issue.

**Comment by kytrinyx** [comment]
To clarify this issue--it's just a matter of changing the messaging:

If the iteration number we get back when having submitted is 1, then we're unlocking the next exercise.

**Comment by wadtech** [comment]
I'm not sure if there was anything still outstanding on this issue, but hopefully #257 means it can be closed.

**Comment by kytrinyx** [comment]
Awww, yeah!! Done and done. Thanks :)
